server/resourcemanager/resource_manager.go

```diff
@@ -222,6 +222,15 @@ func (m *manager[T]) create(w http.ResponseWriter, r *http.Request) {
 		targetResource.Spec = m.rh.SetID(targetResource.Spec, m.config.idgen())
 	}
 
+	if err := targetResource.Spec.Validate(); err != nil {
```

Comment:
I added this validate step to the resource manager, as it wasn't currently in use, not sure if I should better move this to another PR